internal/dag/cache.go

```diff
@@ -100,6 +102,27 @@ func (kc *KubernetesCache) matchesIngressClass(obj metav1.Object) bool {
 
 }
 
+// matchesGateway returns true if the given Kubernetes object
+// belongs to the Gateway that this cache is using.
+func (kc *KubernetesCache) matchesGateway(obj metav1.Object) bool {
```

Comment:
I'd change this function to take a `*gatewayapi_v1alpha1.Gateway` as a parameter. Or, if the intent is to have this work for `HTTPRoutes`/etc. in the future, then I'd probably add a type switch within the implementation, since comparing NS+Name is only valid for `Gateways`.

Reply:
Yup the intent was just to filter out for a specific Gateway. I'll update.